include/links-model.php

```diff
@@ -119,8 +119,8 @@ public function pll_after_languages_cache( $languages ) {
 				$this->set_home_url( $language );
 			}
 
-			// Ensures that the ( possibly cached ) home url uses the right scheme http or https
-			$language->set_home_url_scheme();
+			// Ensures that the ( possibly cached ) home and flag urls use the right scheme http or https.
+			$language->set_url_scheme();
```

Comment:
Actually, the `PLL_Model::get_languages_list()` instantiate the `PLL_Languages` objects, then calls to its `pll_after_languages_cache` filter. (For some reason, [the languages may have already been stored once in the cache](https://github.com/polylang/polylang/blob/7bea3700a7293f8ef31b700fa0fe84647420c53b/include/model.php#L78), but they are also stored **after** this filter has been triggered...)

This triggers the `PLL_Links_Model::pll_after_languages_list()` method which calls the `PLL_Languages::set_url_scheme()` on each language instance.

Given that all this takes place at the time the `PLL_Language` objects are instantiated, I would find it clearer to read if it happened in the `PLL_Language` constructor.

Also, the `set_flag()` method seems to be systematically called each time the `PLL_Language` instances are created (see this [search](https://github.com/polylang/polylang/search?q=set_flag&unscoped_q=set_flag) ), this could be merged in the constructor as well.

Reply:
For `PLL_Languages::set_url_scheme()`, yes we are obliged to do like this, because urls are stored in the transient for both frontedn and amdin, and I have seen some sites with `http` on front and `https` on admin. That's probably less frequent now but I don't want to take the risk.
For `PLL_Languages::set_flag()`, probaby several modifications leaded to this situation.

Reply:
> For `PLL_Languages::set_url_scheme()`, yes we are obliged to do like this, because urls are stored in the transient for both frontedn and amdin, and I have seen some sites with `http` on front and `https` on admin. That's probably less frequent now but I don't want to take the risk.

I still don't get it, but I'll pass.